concordium-consensus/src/Concordium/GlobalState/Basic/TreeState.hs

```diff
@@ -292,8 +292,15 @@ instance (bs ~ BlockState m, BS.BlockStateStorage m, Monad m, MonadIO m, MonadSt
                   else return TS.ObsoleteNonce
           Just (_, Finalized{}) -> return TS.ObsoleteNonce
           Just (tr', results) -> do
+            -- Get the cached verification result if such one is available.
+            -- That is the case if the transaction has status of either `Received` or `Committed`.
```

Comment:
What I wanted from documentation is a reason why this is sensible. I can see from the function definition it does that.

And the reason it is sensible is that in fact the Finalized case is not reachable at all!

Reply:
Updated the doc.